baseAST error on calling function in sub-class

This project is overly-complicated at the moment.  I have checked in a version that should compile with instructions on how to break it.

This is using the `baseAST-error` branch on [Relch](https://github.com/Deep6AI/Relch/tree/feature/baseAST-error)

It relies on several other packages, so you'll need a file called `local.mk` that looks something like

```
CHINGON_HOME=$(HOME)/git/chingon
NUMSUCH_HOME=$(HOME)/git/numsuch
CDO_HOME=$(HOME)/git/cdo
CDOEXTRAS_HOME=$(HOME)/git/cdo-extras
CHARCOAL_HOME=$(HOME)/git/Charcoal
BLAS_HOME=/usr/include/
CHREST_HOME=$(HOME)/git/chrest
RELCH_HOME=$(HOME)/git/Relch
```

Make sure to grab the `develop` branch on all of those.   Once that is in place, go to the top level directory and do `make test`  This should output something like 

```
/home/buddha/git/chapel/third-party/re2/install/linux64-gnu-native/lib -Wl,-rpath,/home/buddha/git/chapel/third-party/re2/install/linux64-gnu-
native/lib   -o /tmp/chpl-buddha-26335.deleteme/test.tmp -L/home/buddha/git/chapel/lib/linux64/gnu/arch-native/loc-flat/comm-none/tasks-qthrea
ds/tmr-generic/unwind-none/mem-jemalloc/atomics-intrinsics/hwloc/re2/fs-none /tmp/chpl-buddha-26335.deleteme/test.tmp.o  /home/buddha/git/chap
el/lib/linux64/gnu/arch-native/loc-flat/comm-none/tasks-qthreads/tmr-generic/unwind-none/mem-jemalloc/atomics-intrinsics/hwloc/re2/fs-none/mai
n.o  -lchpl -lblas -lgmp -lm -lgmp -lchpl -lqthread -L/home/buddha/git/chapel/third-party/hwloc/install/linux64-gnu-native-flat/lib -L/home/bu
ddha/git/chapel/third-party/jemalloc/install/linux64-gnu-native/lib -ljemalloc -lhwloc -lm -lre2 -lpthread
rm -f test/test
mv /tmp/chpl-buddha-26335.deleteme/test.tmp test/test
make[1]: Leaving directory '/home/buddha/git/Relch'
 ** testing RunDefault..we have an FTP
FTP
FTP choice: 3: 0.0 choice: 0 0 1 0 0

0
we have an FTP
FTP
FTP choice: 3: 0.0 choice: 0 0 1 0 0
0
we have an FTP
FTP
FTP choice: 3: 0.0 choice: 0 0 1 0 0
0
we have an FTP
FTP
FTP choice: 3: 0.0 choice: 0 0 1 0 0
0
we have an FTP
FTP
FTP choice: 3: 0.0 choice: 0 0 1 0 0
0
we have an FTP
FTP
FTP choice: 3: 0.0 choice: 0 0 1 0 0
0
.. done (0.000419s)
 ** Passing: 0  Failing: 0
```

Now, to break it, you have to change the return values of each Policy in the `policies.chpl` file back to their actual returns. They have been tested in the [`testPolicies()`](https://github.com/Deep6AI/Relch/blob/feature/baseAST-error/test/RelchTests.chpl#L218-L271) method so I know they work.  The lines are

https://github.com/Deep6AI/Relch/blob/feature/baseAST-error/src/policies.chpl#L15-L16

https://github.com/Deep6AI/Relch/blob/feature/baseAST-error/src/policies.chpl#L36-L37

https://github.com/Deep6AI/Relch/blob/feature/baseAST-error/src/policies.chpl#L82-L83

https://github.com/Deep6AI/Relch/blob/feature/baseAST-error/src/policies.chpl#L109-L110


When you return those to the actual computed value, run `make test` again and you should get

```
make test

 make test
chpl -M/home/buddha/git/chingon/src -M/home/buddha/git/cdo/src -M/home/buddha/git/numsuch/src -M/home/buddha/git/cdo-extras/src -M/home/buddha
/git/Charcoal/src -Msrc --fast --print-callstack-on-error --print-commands -I/usr/include -I/usr/include//include -I -L/usr/include//lib -lbla
s --devel -o test/test test/RelchTests.chpl ;\
./test/test ; \
rm test/test
internal error: no line number available [baseAST.cpp:316]
/bin/sh: 2: ./test/test: not found
rm: cannot remove 'test/test': No such file or directory
Makefile:21: recipe for target 'test' failed
make: *** [test] Error 1
```

I'm not seeing the internal error. I don't have BLAS installed so get an error during C compilation.

I couldn't get a develop branch for CdoExtras.

If I go into Relch and make, I get
```
$ make
Makefile:1: local.mk: No such file or directory
make: *** No rule to make target `local.mk'.  Stop.
```

Well, because I have nerves of steel, I went into the Chapel source code and recompiled.  That got me to this error

```
src/policies.chpl:11: error: conflicting return type specified for 'f(_mt: _MT, this: Policy, options: _array(unmanaged DefaultRectangularArr(
2,int(64),false,int(64),int(64))), state: _array(unmanaged DefaultRectangularArr(1,int(64),false,int(64),int(64)))) [4182022]: int(64)'
src/policies.chpl:96: error:   overridden by 'f(_mt: _MT, this: FollowTargetPolicy, options: _array(unmanaged DefaultRectangularArr(2,int(64),
false,int(64),int(64))), state: _array(unmanaged DefaultRectangularArr(1,int(64),false,int(64),int(64)))) [4477787]: _array(unmanaged DefaultR
ectangularArr(1,int(64),false,int(64),int(64)))'
/bin/sh: 2: ./test/test: not found
rm: cannot remove 'test/test': No such file or directory
Makefile:21: recipe for target 'test' failed
make: *** [test] Error 1
```

Which is actually quite useful.  I'm afraid to close this ticket as it may be my last memory of @mppf but it probably deserves to be once that pull request is merged.

--- a/compiler/resolution/resolveFunction.cpp
+++ b/compiler/resolution/resolveFunction.cpp
@@ -537,6 +537,8 @@ static void insertUnrefForArrayReturn(FnSymbol* fn) {
           if ((rhsType->symbol->hasFlag(FLAG_ARRAY) == true ||
                rhsType->symbol->hasFlag(FLAG_ITERATOR_RECORD)) &&
               isTypeExpr(call->get(2))             == false) {
+
+            SET_LINENO(call);
             Expr*      rhs       = call->get(2)->remove();
             VarSymbol* tmp       = newTemp(arrayUnrefName, rhsType);
             CallExpr*  initTmp   = new CallExpr(PRIM_MOVE,     tmp, rhs);

